typing/typecore.ml

```diff
@@ -5165,8 +5386,9 @@ and type_expect_
       let to_unify = Predef.type_lazy_t ty in
       with_explanation (fun () ->
         unify_exp_types loc env to_unify (generic_instance ty_expected));
-      let env = Env.add_lock Alloc_mode.global env in
-      let arg = type_expect env mode_lazy e (mk_expected ty) in
+      let env = Env.add_locality_lock Locality.global env in
+      (* the thunk will be called only once, so no linearity lock needed *)
```

Comment:
This is true now, but not in OCaml 5 due to potential races. For now let's type it like it will run many times since it will backwards compatible to be more accurate later.